x-pack/plugins/lens/public/pie_visualization/visualization.tsx

```diff
@@ -122,76 +146,56 @@ export const getPieVisualization = ({
     const sortedColumns: AccessorConfig[] = Array.from(
       new Set(originalOrder.concat(layer.groups))
     ).map((accessor) => ({ columnId: accessor }));
-    if (sortedColumns.length > 0) {
-      sortedColumns[0] = {
-        columnId: sortedColumns[0].columnId,
-        triggerIcon: 'colorBy',
-        palette: paletteService
-          .get(state.palette?.name || 'default')
-          .getCategoricalColors(10, state.palette?.params),
+
+    applyPaletteToColumnConfig(sortedColumns, state, paletteService);
```

Comment:
nitpick: I'm not a big fan of side-effects of this type.

What do you think of something like:

```ts
if( sortedColumns.length ){
  sortedColumns[0] = applyPaletteToColumnConfig(sortedColumns, state, paletteService);
}
```